server/src/main/java/org/opensearch/index/store/lockmanager/RemoteStoreMetadataLockManager.java

```diff
@@ -75,7 +76,7 @@ public void release(LockInfo lockInfo) throws IOException {
         }
     }
 
-    public String fetchLock(String filenamePrefix, String acquirerId) throws IOException {
+    public String fetchLockedMetadataFile(String filenamePrefix, String acquirerId) throws IOException {
         Collection<String> lockFiles = lockDirectory.listFilesByPrefix(filenamePrefix);
         List<String> lockFilesForAcquirer = lockFiles.stream()
```

Comment:
The more functional style to do this (and avoid the intermediate List construction) would be the following, though you would lose the assertion check if that's important:

```java
return lockDirectory.listFilesByPrefix(filenamePrefix).stream()
    .filter(lockFile -> acquirerId.equals(FileLockInfo.LockFileUtils.getAcquirerIdFromLock(lockFile)))
    .map(FileLockInfo.LockFileUtils::getFileToLockNameFromLock)
    .findFirst()
    .orElseThrow(() -> new FileNotFoundException("No lock file found for prefix: " + filenamePrefix + " and acquirerId: " + acquirerId));
```